src/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+set(BINARY ${CMAKE_PROJECT_NAME})
+
+file(GLOB_RECURSE SOURCES LIST_DIRECTORIES true *.h *.cpp)
+
+set(SOURCES ${SOURCES})
+
+add_executable(${BINARY}_run ${SOURCES})
+
+add_library(${BINARY}_lib STATIC ${SOURCES})
```

Comment:
If you decided to do a library, then try to rework your CMakeLists in the right way:
main.cpp - move it to the root,
create in the current dir library
create an exe file in the root using the created library (target_link_library(target_name library_name))